src/components/Tabs/AnnotationTab.tsx

```diff
@@ -0,0 +1,360 @@
+            // so we need to stop event propagation so that the row click event
+            // doesn't fire.
+            event.stopPropagation();
+            setLabelOnId(currentLabelIdx!, record.id, false);
+          }}
+        >
+          <CloseOutlined />
+          <VisuallyHidden>
+            Remove ID {record.id} (track {record.track})
+          </VisuallyHidden>
+        </IconButton>
+      ),
+    },
+  ];
+
+  const tableData: TableDataType[] = useMemo(() => {
+    const dataset = props.dataset;
+    if (currentLabelIdx !== null && dataset) {
+      const ids = annotationData.getLabeledIds(currentLabelIdx);
+      return ids.map((id) => {
+        const track = dataset.getTrackId(id);
+        const time = dataset.getTime(id);
+        return { key: id.toString(), id, track, time };
+      });
+    }
+    return [];
+  }, [annotationData, currentLabelIdx, props.dataset]);
+
+  // Options for the selection dropdown
+  const selectLabelOptions: ItemType[] = labels.map((label, index) => {
+    return {
+      key: index.toString(),
+      label: label.ids.size ? `${label.name} (${label.ids.size})` : label.name,
+    };
+  });
+
+  // A small popup that appears when you press the edit button.
+  const editPopoverContents = (
+    <FlexColumn style={{ width: "250px" }} $gap={10}>
+      <label style={{ gap: "10px", display: "flex", flexDirection: "row" }}>
+        <span>Name</span>
+        <Input
+          value={editPopoverNameInput}
+          onChange={(e) => setEditPopoverNameInput(e.target.value)}
+          ref={editPopoverInputRef}
+        ></Input>
+      </label>
+      <label style={{ gap: "10px", display: "flex", flexDirection: "row" }}>
+        <span>Color</span>
+        <div>
+          <ColorPicker
+            size="small"
+            value={new AntdColor(selectedLabel?.color.getHexString() || "ff00ff")}
+            onChange={(_, hex) => {
+              currentLabelIdx !== null && setLabelColor(currentLabelIdx, new Color(hex as HexColorString));
+            }}
+            disabledAlpha={true}
+          />
+        </div>
+      </label>
+      <FlexRow style={{ marginLeft: "auto" }} $gap={10}>
+        <Button onClick={onClickEditCancel}>Cancel</Button>
+        <Button onClick={onClickEditSave} type="primary">
+          Save
+        </Button>
+      </FlexRow>
+    </FlexColumn>
+  );
+
+  return (
+    <FlexColumnAlignCenter $gap={10}>
+      {/* Top-level annotation edit toggle */}
+      <FlexRow style={{ width: "100%" }} $gap={6}>
+        <AnnotationModeButton
+          type="primary"
+          $color={isAnnotationModeEnabled ? "success" : "default"}
+          style={{ paddingLeft: "10px" }}
+          onClick={() => setIsAnnotationModeEnabled(!isAnnotationModeEnabled)}
+        >
+          {isAnnotationModeEnabled ? <CheckOutlined /> : <TagOutlined />}
+          {isAnnotationModeEnabled ? "Done editing" : "Create and edit"}
```

Comment:
why do these need to be two lines? couldn't it be `<CheckOutlined /> "Done editing" : <TagOutlined /> "Create and edit"? also since it's already it's own component, why not send in `isAnnotationModeEnabled` and have it take care of the conditional rendering

Reply:
I realized after I wrote this that it's a styled component. But from line 239 to 256 could def be its own function within this file if not its own component. especially since there are 4 places you're checking for one state

Reply:
Done!